README.md
# doc2vec
document similarity based on gensim doc2vec


news.txt
怀化什么时候能有自己的马拉松赛都市生活怀化新闻网
你希望怀化有自己的马拉松赛吗
谈谈你的看法2012厦门国际马拉松将于1月7日举行厦门国际马拉松赛创办于2003年由中国田径协会和厦门市政府联合主办由中央电视台和厦门电视台联合直播国内外40余家电视台转播经过几年的发展厦门国际马拉松赛以其较高的竞赛水平广泛的群众参与和丰富的配套活动成为中国最具影响力和国际知名度的马拉松赛事之一2007年12月被国际田联评为国际田联路跑金牌赛事并连续三年获此殊荣
比赛时间每年一月份的第一个星期六2012年比赛时间为1月7日星期六比赛项目男女马拉松42.195公里半程马拉松21.0975公里10公里5公里同时举办海峡两岸城市马拉松邀请赛轮椅半程马拉松12公里轮滑比赛规模约8万人每个项目约2万人报满为止相关比赛办法请参阅各项目竞赛规程详情进厦门国际马拉松官网http://www.xmim.org/cn/网上报名已经开始有想法的朋友赶快啊
11月17日我将去现场报名全程如果报名成功将是我第一次参加马拉松争取跑入3小时内
粉丝示爱逗乐朱芳雨邓华德赛前飞吻献赛场球迷网易体育
不到六点男篮队员们就开始了准备活动球迷一声邓华德喊的邓帅眉开眼笑他一个飞吻抛过去人群热闹的哄笑声立刻蔓延在整个赛场国家队今晚对阵黑山队王仕鹏张博两名队员将缺席比赛
华奥星空6月22日昆山电记者杨宝岩王哲林笑了在他身后不远的球迷区他听到有人喊他的名字那一声整齐的王哲林我爱你让他忍不住低头笑了笑不远处师兄朱芳雨也笑了因为这些可爱的球迷喊着他的名字更大声的示爱显然让朱芳雨享受不已
邓华德也笑了和助理教练简单交流后他走近球迷区那一声邓华德喊的他眉开眼笑他一个飞吻抛过去人群热闹的哄笑声立刻蔓延在整个赛场赛前的训练也似乎少了几许凝重
不到六点男篮队员们就开始了准备活动王治郅等老队员依然是在场地边做恢复训练苏伟周鹏朱芳雨和王哲林陈江华郭艾伦等队员进行着加练当球迷们喊着王哲林扣一个的时候郭艾伦从后面持球想来个飞身暴扣虽然没有成功但是球迷们给他热烈的掌声王哲林则在助理教练杜锋的帮助下进行着中投训练国家队今晚对阵黑山队王仕鹏张博两名队员将缺席比赛其他队员里年轻队员依然是邓华德的考察对象完
六成日本民众坚信可夺10金足球体操游泳最热门网易体育
在奥运开幕前的调查中6成日本人认为至少获得10块金牌的最多平均认为会得到8块金牌数
据日媒日刊体育报道24日日本朝日集团的顾客生活文化研究所在11日17日之间通过网络对全国20岁以上的男性和女性共3379人进行了民意调查活动
其中6成日本人认为至少获得10块金牌的最多平均认为会得到8块金牌数男子体操队的内存航平和游泳队的北岛康介被认为夺冠系数最高日本女足的支持率也相当高根据上届北京奥运会的9块金牌作为依据此次伦敦之行至少会拿到10块70年代的男性认为游泳拿到2块女子摔跤拿到2块柔道拿到4块男子体操2块
男性最关注的项目排名第一的是足球65.9第二位游泳53.9第三位体操51.1女性关注项目为第一体操58.0第二足球56.0第三游泳49.4
伦敦奥运女足今晚开战东道主将在主场战新西兰网易体育
上赛季英格兰女超联赛冠军阿森纳女足总共有5人入选成为国脚大户而英国女足最大的优势在于她们的攻击线去年英格兰女超头两号射手威廉姆斯和利特尔都将为国征战新西兰队虽四次参加奥运会但却未尝一胜
时报讯尽管北京时间7月28日凌晨伦敦奥运会开幕式才会掀开神秘面纱但从今晚开始女子足球比赛将率先拉开奥运比赛的大幕女足揭幕战将在东道主英国队与新西兰队之间展开
本次奥运会女足共有12支球队参加分成三个小组这是史上第一支英国女足目前她们基本保持着去年女足世界杯八强英格兰队的班底现在加入几名苏格兰球员后更是实力大增如发挥出色英国女足甚至能给小组最强对手巴西队带来大麻烦
上赛季英格兰女超联赛冠军阿森纳女足总共有5人入选成为国脚大户而英国女足最大的优势在于她们的攻击线去年英格兰女超头两号射手威廉姆斯和利特尔都将为国征战新西兰队虽四次参加奥运会但却未尝一胜
不过唯一让中国球迷遗憾的是本届奥运无论男足还是女足都没有中国队的身影
墨西哥国奥vs韩国前瞻韩国寄诚庸具滋哲首发网易体育
北京时间7月26日21452012年伦敦奥运会男足b组首轮墨西哥国奥队与韩国国奥队的比赛将在纽卡斯尔的圣詹姆斯公园球场进行
北京时间7月26日2145当地时间14452012年伦敦奥运会男足b组首轮墨西哥国奥队与韩国国奥队的比赛将在纽卡斯尔的圣詹姆斯公园球场进行
韩国国奥队在2012年奥运足球亚洲区预选赛中以a组头名的成绩出现他们3胜3平保持不败
墨西哥国奥队在2012年奥运足球中北美及加勒比海地区预选赛中以5全全胜的成绩入围正赛他们在半决赛中31击败加拿大国奥队后又在决赛中21战胜洪都拉斯国奥队
韩国国奥队希望改写自己在奥运会足球比赛的战绩他们曾在1948年伦敦奥运会和2004年雅典奥运会中2次打进8强
墨西哥国奥队曾在1996年亚特兰大奥运会足球比赛中小组出线打进8强此外他们还参加过1992年巴塞罗那奥运会和2004年雅典奥运会但均未能小组出线
1948年伦敦奥运会中两队曾在第一轮比赛中有过交手记录当时韩国队53战胜了墨西哥队
奥运会禁止女柔道选手戴头巾参赛沙特扬言退赛网易体育
很长时间以来我们对于伊斯兰宗教国家的女子第一印象就是她们的头巾不过在伦敦奥运会上将可能有一位沙特阿拉伯的女子柔道选手打破我们这种印象她就是沃丹・沙赫卡尼
由于当地的宗教信仰沙特是一直以来唯一一个从来不派女运动员参加奥运会的国家伦敦奥运会是他们第一次派女运动员参赛这件事情本身就非常曲折直到开赛几周前国际奥委会才宣布有两名女沙特运动员将出现在赛场上分别是参加柔道的沙赫卡尼和田径的莎拉-阿塔与此同时沙特人强调只有国际奥委会同意沙特女将身着传统穆斯林服装参赛他们才能同意放行
随后这件事情再现反复根据美国雅虎的报道为了保证运动员在比赛中的安全国际柔道联合会对沙特的运动员提出要求沙赫卡尼必须摘去头巾比赛因为在柔道比赛中头上佩戴任何饰物都可能成为安全隐患根据比赛规定队员必须赤足穿柔道衣比赛因为如果再比赛中间一旦对手抓住了沙赫卡尼的头巾很容易造成头部和颈部受伤
于是沙特不干了英国著名的每日电讯报曝料国际奥委会沙特奥委会以及国际柔道联合会和伦敦奥组委四方召开了紧急会议因为沙特对于国际柔道联合会的做法非常不满他们表示要么沙赫卡尼带头巾出战要么沙特就退出奥运会
羽球女单李雪芮率先亮相横扫对手获小组开门红网易体育
伦敦奥运会羽毛球女单比赛首日位于l组的李雪芮在中国女单三名选手中率先亮相面对实力远逊自己的秘鲁选手里维罗直落两局以215和216获胜获得小组赛开门红
李雪芮今年自尤伯杯亚洲区预选赛起一度连胜30多场拿下包括全英赛在内的多项冠军凭借着几个月来极为出色的表现在最后时刻力压王适娴入围奥运名单后来居上成为名符其实的黑马
作为三号种子李雪芮成为伦敦奥运会中国女单最先出场的选手被分在下半区的她将力争与汪鑫会师决赛小组赛首战李雪芮对阵实力较弱的秘鲁选手里维罗不费吹灰之力就拿下对手
比赛开始李雪芮一上场就打出气势很快占据场上主动速度上明显上风进攻更是占据压倒性优势对手根本跟不上她的节奏经过短暂的开场适应后李雪芮不断扩大优势对手根本无法抵挡她的攻势李雪芮以215轻松大胜
第二局李雪芮继续掌控场上主动双方实力差距明显李雪芮拉吊突击很好地控制住落点迅速拉开分差打得里维罗毫无还手之力迅速拉开分差暂停时已经以113领先暂停后李雪芮优势更为明显双方完全不在同一水平线上李雪芮越战越勇对手则失误不断李雪芮以216再下一城直落两局轻取对手赢得小组赛首胜
邓华德三大赛屡屡涉险过关福帅能否延续好运网易体育
2010年在男篮世锦赛小组赛中中国队只赢了一场比赛但最终以小分0.018分的优势跻身16强为此邓华德有了福帅之誉伦敦奥运会希望福帅延续好运
在不久前公布的伦敦奥运会中国代表团名单中有11位外籍教练的名字分别出现在田径游泳击剑自行车篮球曲棍球赛艇等8个项目中这些洋帅不但带来了独特的训练技术和理念也为中国队在伦敦奥运争金夺银增添希望
2010年中国篮协宣布美国人邓华德出任中国男篮主教练邓华德上任以来经历过三次大考分别是男篮世锦赛亚运会和亚锦赛这三次大赛中国男篮都没有姚明亚运会时甚至连易建联也缺席邓华德就是带领这样一支中国男篮过关斩将
2010年在男篮世锦赛小组赛中中国队只赢了一场比赛但最终以小分0.018分的优势跻身16强2010年广州亚运会中国队无姚无易但总是在生死关头保住几乎溜走的胜利2011年亚锦赛此前小组赛全胜场均能赢对手45.5分的伊朗队却被约旦队淘汰决赛中中国队又以1分优势险胜约旦从而提前获得伦敦奥运会入场券为此邓华德有了福帅之誉伦敦奥运会希望福帅延续好运
易建联正式归队参加合练邓华德
易建联今天正式归队并且与男篮进行了首次合练易建联在今天接受记者采访时表示目前自己的状态是所参加的几届奥运会以来最好的一次
昨天刚刚结束拉练抵达北京的男篮今天照常在训练局训练而核心易建联也在今天正式归队并且与全队进行了合练易建联在今天接受记者采访时表示目前自己的状态是所参加的几届奥运会以来最好的一次
自从男篮本次集中以来易建联一直都没有归队目前距离奥运会只剩下一个多月时间而刚刚结束了澳洲拉练的中国男篮以全败成绩收场这个时候易建联的加盟对于男篮来说是件再好不过的消息了就在易建联今天首次与全队合练的时候主帅邓华德还为阿联拍了一张投篮的背影照片表示猜猜谁回来了这对中国队来说是一个好消息
在男篮进行热身赛的同时易建联也没有放松一直在训练师的陪伴下坚持着训练易建联在今天接受记者采访时表示目前自己的状态是所参加的几届奥运会以来最好的一次并且易建联自信地称现在的中国男篮早已适应了没有姚明的打法
6月22日男篮四国赛将开打易建联也会在这一轮热身赛上登场亮相
国际体联发布新规则伦敦奥运将可产生并列冠军网易体育
国际体操联合会昨日向外界发布公告称将重新在比赛中采用打分评判中的平分规则这意味着在伦敦的体操赛场上团体和个人都可能再次出现并列冠军
国际体操联合会近日向外界发布公告称打分评判中的平分规则在阔别大概15年后重新回归这意味着在伦敦的体操赛场上团体和个人都可能再次出现并列冠军
国际奥委会在1996年亚特兰大奥运会后要求国际体联结束在体操比赛中出现平分成绩并列的情况于是国际体联在1997年出台了新的打分规定其中最关键的就是保证一旦出现平分情况仍然可以区分运动员的最终名次
1总分相加后高分者取胜即对于男子依次取最高分的54321个项目的总分比较对于女子依次取最高分的321个项目比较
2如果仍然并列两跳中e分高的一跳的e分高者获胜
3如果仍然并列两跳中d分高的一跳的d分高者获胜
1总分相加后高分者取胜即对于男子依次取最高分的54321个项目的总分比较对于女子依次取最高分的321个项目比较
2013世界政要的喜怒哀乐安徽频道人民网
奥巴马的2013年无疑是在连任的喜悦和对第二个总统任期的憧憬中开始的1月21日正式就职之后其支持率甚至冲到了60%的3年来最高点然而随着控枪的虎头蛇尾奥巴马在美国国内和国际上开始遭遇一系列难题虽然一系列的出访和首脑会晤让奥巴马在外交上取得了一定成果但是随着棱镜门事件的曝光麻烦开始接踵而至叙利亚化武危机联邦政府被迫关门美被曝监听多国领导人等事件让奥巴马应接不暇而其最为看重的医疗改革在年底也是问题频发招致美国各界批评甚至将其支持率推到了41%的低点可以说2013年的奥巴马真是焦头烂额图为2013年6月25日奥巴马在乔治城大学发表演说提出了减少碳排放的未来目标在演讲中奥巴马突然暂停掏出手帕擦拭脸上的汗水
英国全球联网预防动物疾病-农博畜牧
上周英国推出全球科学研究网这一网络的推出更助英国预防一些如禽流感口蹄疫等动物疾病
日益全球化的动物运输感染动物疾病的高风险食品和人群在英国开始蔓延这就有可能导致非常严重的经济环境和健康问题
这个国际网络是由欧盟资助英国环境部门领导全世界数千名科学家参与到其中在这个平台上他们可以相互交换研究树立共同的目标或合作研究对将来一些可能发生的疫情进行控制同时也有助于电子预警系统来判断疫情来自哪个国家
在全球联网实施前农业和食品部长jimpaice在周三发表讲话时说这是一个全球化贸易和世界旅行的时代动物疾病随之进入英国的危险也要高于以前
农博网声明本文仅代表作者本人观点与农博网无关其原创性以及文中陈述文字和内容未经本站证实对本文以及其中全部或者部分内容文字的真实性完整性及时性本站不作任何保证或承诺请读者仅作参考并请自行核实相关内容
通用印度陷排放测试丑闻面临巨额罚款汽车环球网
环球网综合报道据美国媒体leftlanenews10月22日报道印度政府调查报告得出结论7月份雪佛兰tavera多用途车在印度爆出的排放测试丑闻的罪魁祸首正是通用汽车公司将面临最高1亿卢比约合160万美元的罚款
一位拒绝透露姓名的政府官员告诉路透社这份报告指出事件实质是公司欺诈报告显示原设备制造商通用公司应当对所发生的一切负全责印度测试实验室并无过错
今年7月份通用公司向印度政府承认其员工在排放测试中使用不正当手段据内部消息员工们为参加测试的雪佛兰安装了更清洁的发动机并且为避免更严厉的车型分级谎报了该车的重量
这一事件迫使通用停产雪佛兰taera并召回11.4万辆该车稍后公司全球发动机工程副总裁也因违反公司准则被辞退实习编译盛苗审稿刘洋
奥巴马讲述与妻子第一次约会细节自曝追妻攻略图
据英国每日电讯报6月25日报道美国总统奥巴马日前在其网站上发布了一段视频详细讲述了如何在与妻子米歇尔第一次约会时就令她为自己着迷
在这段视频中奥巴马和米歇尔共同回忆了1989年两人在芝加哥第一次约会的细节他们共度了一天的浪漫时光前往艺术馆参观共进午餐散步还观看了一场电影
米歇尔在视频中说这次约会棒极了我们一整天都在一起他向我展示了他性格的各个方面我们去了艺术馆这时奥巴马指着镜头外说艺术小伙子们记住了艺术会给人留下好印象
米歇尔继续说他展示了他的文化修养我们沿着密歇根大街愉快地散步奥巴马插话说这是浪漫
随后奥巴马谈起两人在艺术馆的庭院里同进午餐还特别提到院内的喷泉米歇尔补充说他展示了每一面时髦聪明修养敏感喷泉显示了他的感性散步证明他有耐心
奥巴马夫妇是在米歇尔工作的法律公司第一次见面的三年后也就是1992年他们就结了婚这并不是美国总统夫妇第一次与民众分享浪漫经验此前接受采访时他们曾强调经常约会对已婚夫妇的重要性
图集硬汉普京铁血柔情老婆年轻时超级美女银行频道证券之星

doc2vec.py
#! -*- coding: utf-8 -*-
import jieba
from gensim.models.doc2vec import Doc2Vec, TaggedDocument

stop_text = open('stop_words.txt', 'r', encoding='utf-8')
stop_words = set([i.strip('\n') for i in stop_text.readlines()])


def cut_without_stop_words(text):
    return [i for i in jieba.cut_for_search(text) if i not in stop_words]


def get_corpus():
    with open("news.txt", 'r', encoding='utf-8') as doc:
        docs = doc.readlines()
    train_docs = []
    for i, text in enumerate(docs):
        word_list = cut_without_stop_words(text)
        length = len(word_list)
        word_list[length - 1] = word_list[length - 1].strip()
        document = TaggedDocument(word_list, tags=[i])
        train_docs.append(document)
    return train_docs


def train(x_train, size=200, epoch_num=70):
    model_dm = Doc2Vec(x_train, min_count=1, window=3, size=size, sample=1e-3, negative=5, workers=4)
    model_dm.train(x_train, total_examples=model_dm.corpus_count, epochs=epoch_num)
    model_dm.save('model_doc2vec')
    return model_dm


def predict(sentence, top_k=10):
    model_dm = Doc2Vec.load("model_doc2vec")
    text_cut = cut_without_stop_words(sentence)
    inferred_vector_dm = model_dm.infer_vector(text_cut)
    sim_n = model_dm.docvecs.most_similar([inferred_vector_dm], topn=top_k)
    return sim_n


if __name__ == '__main__':
    x_train = get_corpus()
    model_dm = train(x_train)
    sims = predict(sentence='经现场相关人员了解雨天路滑视线模糊加之车辆速度过快导致了该起事故的发生')
    for count, sim in sims:
        sentence = x_train[count]
        words = ''
        for word in sentence[0]:
            words = words + word + ' '
        print(sim, len(sentence[0]), words)


d2v.py
#! -*- coding: utf-8 -*-
import os
from multiprocessing import cpu_count

import jieba
from gensim.models.doc2vec import Doc2Vec, TaggedDocument

stop_text = open('stop_words.txt', 'r', encoding='utf-8')
stop_words = set([i.strip('\n') for i in stop_text.readlines()])
jieba.load_userdict('user_dict.txt')


def cut_without_stop_words(text):
    return [i for i in jieba.cut_for_search(text) if i not in stop_words]


def get_corpus():
    with open("news.txt", 'r', encoding='utf-8') as doc:
        docs = doc.readlines()
    train_docs = []
    for i, text in enumerate(docs):
        word_list = cut_without_stop_words(text)
        length = len(word_list)
        word_list[length - 1] = word_list[length - 1].strip()
        document = TaggedDocument(word_list, tags=[i])
        train_docs.append(document)
    return train_docs


def train(x_train, size=200, epoch_num=70):
    model_dm = Doc2Vec(x_train, seed=321, min_count=1, window=5, vector_size=size, sample=1e-3, negative=5, workers=cpu_count())
    model_dm.train(x_train, total_examples=model_dm.corpus_count, epochs=epoch_num)
    model_dm.save('model_doc2vec')
    return model_dm


def predict(sentence, top_k=10):
    model_dm = Doc2Vec.load("model_doc2vec")
    text_cut = cut_without_stop_words(sentence)
    inferred_vector_dm = model_dm.infer_vector(text_cut)
    sim_n = model_dm.docvecs.most_similar([inferred_vector_dm], topn=top_k)
    return sim_n


if __name__ == '__main__':
    x_train = get_corpus()
    if not os.path.exists('model_doc2vec'):
        model_dm = train(x_train)
    sims = predict(sentence='经现场相关人员了解雨天路滑视线模糊加之车辆速度过快导致了该起事故的发生')
    for count, sim in sims:
        sentence = x_train[count]
        words = ''
        for word in sentence[0]:
            words = words + word + ' '
        print(sim, len(sentence[0]), words)
